v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/DataStreamToSpannerCDCFT.java

```diff
@@ -0,0 +1,239 @@
+            pubsubResourceManager,
+            dlqGcsPrefix,
+            gcsResourceManager);
+    String artifactBucket = TestProperties.artifactBucket();
+
+    // launch datastream
+    DatastreamResourceManager.Builder datastreamResourceManagerBuilder =
+        DatastreamResourceManager.builder(testName, PROJECT, REGION)
+            .setCredentialsProvider(credentialsProvider);
+    if (System.getProperty("privateConnectivity") != null) {
+      datastreamResourceManagerBuilder.setPrivateConnectivity(
+          System.getProperty("privateConnectivity"));
+    }
+    datastreamResourceManager = datastreamResourceManagerBuilder.build();
+    Stream stream =
+        createDataStreamResources(
+            artifactBucket, gcsPrefix, sourceConnectionProfile, datastreamResourceManager);
+
+    int numRows = 100;
+    int burstIterations = 10000;
+
+    // generate Load
+    cdcLoadGenerator = new FuzzyCDCLoadGenerator();
+    cdcLoadGenerator.generateLoad(numRows, burstIterations, 0.5, sourceDBResourceManager);
+
+    FlexTemplateDataflowJobResourceManager.Builder flexTemplateBuilder =
+        FlexTemplateDataflowJobResourceManager.builder(testName)
+            .addEnvironmentVariable("numWorkers", NUM_WORKERS)
+            .addEnvironmentVariable("maxWorkers", MAX_WORKERS);
+
+    // launch dataflow template
+    jobInfo =
+        launchFwdDataflowJob(
+            spannerResourceManager,
+            gcsPrefix,
+            stream.getName(),
+            dlqGcsPrefix,
+            subscription.toString(),
+            dlqSubscription.toString(),
+            flexTemplateBuilder,
+            shadowTableSpannerResourceManager);
+  }
+
+  /**
+   * Cleanup all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  @After
+  public void cleanUp() throws IOException {
+    ResourceManagerUtils.cleanResources(
+        spannerResourceManager,
+        sourceDBResourceManager,
+        datastreamResourceManager,
+        gcsResourceManager,
+        pubsubResourceManager);
+  }
+
+  @Test
+  public void liveMigrationCrossDbTxnCdcTest()
+      throws IOException, InterruptedException, SQLException, ExecutionException {
+
+    // Wait for Forward migration job to be in running state
+    assertThatPipeline(jobInfo).isRunning();
+
+    // Wait for at least 1 row to appear in Spanner
+    ConditionCheck conditionCheck =
+        SpannerRowsCheck.builder(spannerResourceManager, USERS_TABLE).setMinRows(1).build();
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(20)), conditionCheck);
+    assertThatResult(result).meetsConditions();
+
+    // Kill the dataflow workers multiple times to induce work item assignment rebalancing and
+    // inturn increase the chance of same key being processed by multiple workers parallelly.
+    DataflowFailureInjector.abruptlyKillWorkers(jobInfo.projectId(), jobInfo.jobId());
```

Comment:
it would make sense to keep killing workers every 20-30 seconds until the end of the job. Any reason to stop after 3 iterations ?